### PR TITLE
feat: [CDS-42038]:fs-configs-exec

### PR DESCRIPTION
**Pull request by litvinovstas**
### Summary

execution strategy for winrm deployment type improves, 
resolved issue with file store default tab and error message display

#### Screenshots
Screenshot 2022-09-06 at 09 29 52
Screenshot 2022-09-06 at 09 34 21

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

## [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 6 Issues 
* Code issue 6 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11008)

File: src/modules/31-filestore/components/MultiTypeFileSelect/FileStoreSelect/FileStoreSelectField.tsx [hunk 1]
```diff
@@ -87,6 +77,17 @@ function FileStoreInput(props: FormikFileStoreInput): React.ReactElement {
         }
     }
   }
+  const modalFileStore = useFileStoreModal({
+    applySelected: value => {
+      const { scope, path } = value
+      onChange?.(prepareFileStoreValue(scope, path))
```
Comment (chetanvekariya-harness) [inline src/modules/31-filestore/components/MultiTypeFileSelect/FileStoreSelect/FileStoreSelectField.tsx:83]: Call prepareFileStoreValue only once and store returned value in variable, then reuse returned value.

File: src/modules/70-pipeline/components/PipelineStudio/ExecutionStrategy/__tests__/ExecutionStrategy.test.tsx [hunk 2]
```diff
@@ -612,4 +612,45 @@ describe('ExecutionStrategy test', () => {
...
+
+    const packageSelect = container.querySelector('input[name="packageType"]')
+    expect(packageSelect).toBeInTheDocument()
+
+    await act(async () => {
+      fireEvent.change(packageSelect!, {
+        target: { value: 'ISS_APP' }
+      })
+    })
+    expect(packageSelect!).toHaveValue('common.application')
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineStudio/ExecutionStrategy/__tests__/ExecutionStrategy.test.tsx:654]: Shouldn't value be `ISS_APP` ?

**Review by chetanvekariya-harness — APPROVED** [review]

**Comment by YounixIO** [comment]
@litvinovstas : Please add Link to the Coverage task as link to the PR.

**Review by radujomir — DISMISSED** [review]

File: src/modules/31-filestore/components/MultiTypeFileSelect/FileStoreSelect/FileStoreSelectField.tsx [hunk 3]
```diff
@@ -80,7 +80,8 @@ function FileStoreInput(props: FormikFileStoreInput): React.ReactElement {
   const modalFileStore = useFileStoreModal({
     applySelected: value => {
       const { scope, path } = value
-      onChange?.(prepareFileStoreValue(scope, path))
+      const preparedValue = prepareFileStoreValue(scope, path)
+      onChange?.(preparedValue)
       formik.setFieldValue(name, prepareFileStoreValue(scope, path))
```
Comment (chetanvekariya-harness) [inline src/modules/31-filestore/components/MultiTypeFileSelect/FileStoreSelect/FileStoreSelectField.tsx:85]: Use `preparedValue` here as well. There are two calls happening even right now.

File: src/modules/70-pipeline/components/PipelineStudio/ExecutionStrategy/__tests__/ExecutionStrategy.test.tsx [hunk 4]
```diff
@@ -648,9 +648,9 @@ describe('ExecutionStrategy test', () => {
 
     await act(async () => {
       fireEvent.change(packageSelect!, {
-        target: { value: 'ISS_APP' }
+        target: { value: 'common.application' }
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineStudio/ExecutionStrategy/__tests__/ExecutionStrategy.test.tsx:651]: This means that when you try to select value `ISS_APP`, it is not working. And that is why you might have gone back to `common.application`. We should rather try to write test case where in you select value `ISS_APP`. I feel common.application is default value and that is why test case is passing.

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**